std/stdio.d

```diff
@@ -5457,18 +5455,59 @@ private struct ReadlnAppender
     }
 }
 
-// Private implementation of readln
-private size_t readlnImpl(FILE* fps, ref char[] buf, dchar terminator, File.Orientation orientation)
+private struct LockedFile
 {
-    version (DIGITAL_MARS_STDIO)
+    private @system FILE* fps;
+
+    this(FILE* fps) @trusted
     {
+        this.fps = fps;
         _FLOCK(fps);
-        scope(exit) _FUNLOCK(fps);
+    }
+
+    @disable this();
+    @disable this(this);
+    @disable void opAssign(LockedFile);
+
+    // Since fps is now locked, we can cast away shared
+    @trusted fp() return scope => cast(_iobuf*) fps;
```

Comment:
This method should be `@system`

Reply:
I've removed this method now. I assume you said that because `return scope` doesn't enforce that the result doesn't live longer than the struct. That seems to be a significant missing piece with -dip1000.

Reply:
I mean the point is to encapsulate the locked pointer away from usage.

I didn't realize you weren't actually storing the unlocked pointer, but using this to cast it. I was expecting the `LockedFile` to be storing the locked pointer, and protect it from incorrect usage.